### PR TITLE
Add vault configuration to quick-start.

### PR DESCRIPTION
**Pull request by revital76**
This PR adds the following Vault configuration to the quick-start:

1) it configures Kubernetes auth method in path called kubernetes: this auth path is used later to add roles for the modules and secret provider.
2) it enables Kubernetes secret plugin reader
3) it creates a policy to access secrets
4) it creates a role to bind modules running in m4d-blueprint namespace to use the policy above.
5) it creates userpass auth method.

This configuration completes the existing Vault configuration done in the manager:
1) it mounts secret engine for dataset credentials
2) it adds a policy to access it
3) it adds a role in kubernetes auth path for the secret provider to use the policy above

### CODE REVIEW

File: third_party/vault/vault-single-cluster/values.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+server:
```
Comment (roee88) [inline third_party/vault/vault-single-cluster/values.yaml:1]: Can you create a separate PR in upstream to add just this file?

File: third_party/vault/vault-single-cluster/values.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  volumes:
+    - name: plugins
+      emptyDir: {}
+  volumeMounts:
+    - mountPath: /usr/local/libexec/vault
+      name: plugins
+      readOnly: false
+  extraInitContainers:
+    - name: kubesecrets-plugin
+      image: ghcr.io/revit13/vault-plugin-secrets-kubernetes-reader:latest
```
Comment (roee88) [inline third_party/vault/vault-single-cluster/values.yaml:12]: Why is it using ghcr.io/revit13?

Comment (revit13) [reply third_party/vault/vault-single-cluster/values.yaml:12]: by mistake... will fix that.

File: third_party/vault/vault-single-cluster/values.yaml (hunk 1 above)
Comment (revit13) [reply third_party/vault/vault-single-cluster/values.yaml:1]: Sure, will do that

Comment (revit13) [reply third_party/vault/vault-single-cluster/values.yaml:1]: @roee88 should I add also the vault-rbac.yaml file to this PR?

Comment (roee88) [reply third_party/vault/vault-single-cluster/values.yaml:1]: No

File: site/docs/get-started/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    kubectl create namespace m4d-system || true
+    kubectl apply -f https://raw.githubusercontent.com/IBM/the-mesh-for-data/master/third_party/vault/vault-single-cluster/vault-rbac.yaml -n m4d-system
+    helm repo add hashicorp https://helm.releases.hashicorp.com
+    helm install vault hashicorp/vault --version 0.9.1 \
+        --set "global.openshift=true" \
+        --set "server.dev.enabled=true" \
+        --values https://raw.githubusercontent.com/IBM/the-mesh-for-data/master/third_party/vault/vault-single-cluster/values.yaml \
+        --namespace m4d-system \
+        --wait --timeout 120s
+    ```
```
Comment (revit13) [inline site/docs/get-started/quickstart.md:59]: @roee88 this part needs to be included in the quick-start...

Comment (roee88) [reply site/docs/get-started/quickstart.md:59]: I know :)

**Pull request closed without merging**